### ISSUE TITLE
Multiple attempts with the same log_path

### ISSUE THREAD

**Issue by cgardens** [airbytehq/airbyte#1343]
## Expected Behavior
* Each attempt should have its own log path.

## Current Behavior
* In one case two attempts were created with the same log path. It seems likely it is related to this behavior specified in a [comment in the code](https://github.com/airbytehq/airbyte/blob/master/airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java#L94-L98).
* Setting this to high urgency because there is a lot of undefined behavior that crops up when this happens.

## Steps to Reproduce
* Have not yet been able to reproduce it. It should not be possible to create an attempt for a job while it already has a running attempt.
* Evidence that it happened:
```
airbyte=# SELECT attempts.* FROM jobs LEFT OUTER JOIN attempts ON jobs.id = attempts.job_id where jobs.id=48;
 id | job_id | attempt_number |           log_path           | output | status  |          created_at           |          updated_at           | ended_at
----+--------+----------------+------------------------------+--------+---------+-------------------------------+-------------------------------+----------
 48 |     48 |              0 | /tmp/workspace/48/0/logs.log |        | failed  | 2020-12-15 22:34:02.793586+00 | 2020-12-15 22:34:09.056187+00 |
 50 |     48 |              2 | /tmp/workspace/48/1/logs.log |        | running | 2020-12-15 22:36:23.387997+00 | 2020-12-15 22:36:23.387997+00 |
 49 |     48 |              1 | /tmp/workspace/48/1/logs.log |        | failed  | 2020-12-15 22:35:13.094404+00 | 2020-12-15 23:37:26.446024+00 |
(3 rows)
```
* It seems like between this [line](https://github.com/airbytehq/airbyte/blob/master/airbyte-scheduler/src/main/java/io/airbyte/scheduler/WorkerRunFactory.java#L82) and this [one](https://github.com/airbytehq/airbyte/blob/master/airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java#L99) another attempt was created for the job
* Things to check: 
  * can multiple job submitters run at once (this should not be the case but possible we do do this right)
  * possible fix would be to pass the attempt number into the job run factory.
  * i can't get the old schedule / submitter logs to see if we logged anything useful when this error occurred.

**Comment by jrhizor** [comment]
I also ran into this.

**Comment by cgardens** [comment]
@jrhizor - damn. you have any clues on how to reproduce?

**Comment by cgardens** [comment]
Had thought that this was a race condition that was fixed by PR to prevent multiple jobs from running at once. That wasn't true. New pr up to fix this problem directly #1430
